src/test/java/com/contentgrid/gateway/runtime/session/PartitionedWebSessionManagerTest.java

```diff
@@ -0,0 +1,149 @@
+                .returnResult(Void.class).getResponseCookies().getFirst(SESSION_COOKIE_NAME);
+
+        assertThat(sessionCookie).isNotNull();
+        assertThat(sessionCookie.getValue()).isNotEmpty();
+
+        // HttpOnly: session cookie should not be accessible to javascript
+        assertThat(sessionCookie.isHttpOnly()).isTrue();
+        // Secure: only send session cookie over HTTPS
+        assertThat(sessionCookie.isSecure()).isTrue();
+        // Max-Age: Negative Max-Age means cookie is deleted when the browser session ends
+        assertThat(sessionCookie.getMaxAge()).isNegative();
+        // Domain: specifying "Domain" is less restrictive, because that also implies subdomains
+        assertThat(sessionCookie.getDomain()).isNull();
+        // SameSite: Lax; cookie is sent in a first-party context or following a link to the origin
+        assertThat(sessionCookie.getSameSite()).isEqualTo("Lax");
+    }
+
+    @Test
+    void sessionCookie_resumingSession() {
+
+        var sessionCookie = client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentgrid.app/test")
+                .exchange()
+                .expectStatus().isOk()
+                .expectCookie().exists(SESSION_COOKIE_NAME)
+                .expectBody().jsonPath("$.previous-request-id").isEqualTo("null")
+                .returnResult().getResponseCookies().getFirst(SESSION_COOKIE_NAME);
+
+        assertThat(sessionCookie).isNotNull();
+
+        // request including the session cookie to the same domain
+        // should NOT rotate session id
+        client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentgrid.app/test")
+                .cookie("SESSION", sessionCookie.getValue())
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().doesNotExist("Set-Cookie")
+                .expectBody().jsonPath("$.previous-request-id").isEqualTo("1");
+    }
+
+
+    @Test
+    void sessionCookie_differentDomains() {
+        var sessionCookie = client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://app.userapps.contentgrid.app/test")
+                .exchange()
+                .expectStatus().isOk()
+                .expectCookie().exists(SESSION_COOKIE_NAME)
+                .returnResult(Void.class).getResponseCookies().getFirst(SESSION_COOKIE_NAME);
+
+        assertThat(sessionCookie).isNotNull();
+
+        // trying to re-use the session cookie for a different hostname
+        client
+                .mutateWith(mockOAuth2Login())
+                .get()
+                .uri("https://other.userapps.contentgrid.app/test")
+                .cookie("SESSION", sessionCookie.getValue())
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().exists("Set-Cookie");
```

Comment:
Needs a check here for previous request id as well.
We expect it to be `null` here, right?